### DISCUSSION TITLE
Type-Driven Development | kciter.so

### DISCUSSION THREAD

**Discussion by giscus** [kciter/kciter.github.io#27] [Announcements]
# Type-Driven Development | kciter.so

프로그래밍 세계에는 문제를 해결하기 위한 다양한 방법론이 존재한다. 크게는 업무를 처리하는 방법부터 작게는 개발자 개인이 코딩하는 방법까지 다양하다. 이 글에서는  타입 시스템을 이용한 방법론인 타입 주도 개발 에 대해 소개한다. 타입은 정적 타입 …

https://kciter.so/posts/type-driven-development

**Comment by ibcylon** [comment]
와우

**Comment by hanbyul-kim** [comment]
본문에서 사용하신 단사함수에 대한 설명이 정확한 것 맞나요?

**Reply by kciter** [reply]
다시 살펴보니 제가 잘못 이해하고 작성한 있는 것 같네요. 이 부분은 오늘 중으로 다시 수정하겠습니다. 지적해주셔서 감사해요 🙂

**Comment by SaeWooKKang** [comment]
우와 글에서 내공이 느껴져요! 혹시 타입주도개발 관련 초심자가 학습할 수 있는 책이나 강연이 있을까요? 👀

**Reply by kciter** [reply]
아쉽게도 제가 조사했을 때 초심자를 위한 책이나 강연은 따로 없었습니다. 😭 그래도 제가 참고했던 자료를 공유해 드리자면,

국내 자료 중에선 [김혜성 님이 발표하신 자료](https://www.youtube.com/watch?v=M3pMCZqPvzI)가 있었고 [Type-Driven Development with Idris에 대한 리뷰](https://harfangk.github.io/2017/10/23/tdd-with-idris-review-ko.html) 글이 있었습니다. 해외 자료 중에선 F#으로 작성된 [Designing with types 시리즈](https://fsharpforfunandprofit.com/series/designing-with-types/)가 유용했습니다.

개인적으로 타입 주도 개발은 타입 시스템에 대한 이해와 이 글에서 작성된 타입에 대한 생각이 중요하다고 생각해서 별도로 타입 주도 개발이라는 것을 공부하기보다 [타입으로 견고하게 다형성으로 유연하게 책](https://product.kyobobook.co.kr/detail/S000210397750)을 읽어보시면 도움이 되실 거로 생각합니다. 😄
